src/Compilers/.editorconfig

```diff
@@ -12,6 +12,15 @@ dotnet_diagnostic.RS0101.severity = none
 # RS0102: Braces must not have blank lines between them
 dotnet_diagnostic.RS0102.severity = none
 
+# IDE0059: Unnecessary assignment to a value
+dotnet_diagnostic.IDE0059.severity = none
```

Comment:
this was unfortunate.  but without this the compiler had tons of warnings here that touched a lot of core code.  i think we should take a followup PR that enables this in isolation.

Reply:
@sharwell @jcouv for thoughts on this.  I really don't like the idea of disabling this (as i think it reveals many interesting cases where the code seems fishy).  But i don't want to have this PR go in witha  bunch of complex code changes that have to be closely reviewed and which carries the risk of a subtle bug being introduced.

Reply:
Is IDE0059 new for 3.10? I throught we disabled it for the compiler code a long time ago.

Reply:
Not sure. @mavasani ?  The diagnostics look appropriate. But there are a lot of them and I don't think we should do this without explicit compiler desire ahead of time.

Reply:
@mavasani any ideas here?

Reply:
> but without this the compiler had tons of warnings

This cannot be correct. Our editorconfig does not set IDE0059 to warning for the compiler folder.